src/NuGet.Clients/NuGet.VisualStudio.Common/ErrorListTableDataSource.cs

```diff
@@ -168,17 +169,30 @@ public void AddNuGetEntries(params ErrorListTableEntry[] entries)
         /// <summary>
         /// Show error window.
         /// </summary>
-        public void BringToFront()
+        public void BringToFrontIfSettingsPermit()
         {
             EnsureInitialized();
 
             NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
             {
                 await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
-                // Give the error list focus.
-                var vsErrorList = _errorList as IVsErrorList;
-                vsErrorList?.BringToFront();
+                IVsShell vsShell = await _asyncServiceProvider.GetServiceAsync<IVsShell>();
+                object propertyShowTaskListOnBuildEnd;
```

Comment:
nit: 
you can do
```cs
int getPropertyReturnCode = vsShell.GetProperty((int)__VSSPROPID.VSSPROPID_ShowTasklistOnBuildEnd, out object propertyShowTaskListOnBuildEnd);
```

Reply:
Yes... I split it out because I find it easier to debug and more human-readable. I can change it if it's violating some coding guideline.

Reply:
I find the other more readable cause it removes redundant lines, but yeah we don't have a rule for it so do whatever until we have one (if ever).